src/python/pants/option/parser.py

```diff
@@ -344,7 +347,8 @@ def add_flag_val(v: Optional[Union[int, float, bool, str]]) -> None:
 
             # If the option is explicitly given, check deprecation and mutual exclusion.
             if val.rank > Rank.HARDCODED:
-                self._check_deprecated(dest, kwargs)
+                print_warning = not parse_args_request.suppress_warnings
+                self._check_deprecated(dest, kwargs, print_warning=print_warning)
```

Comment:
Could you instead only call this method if it's relevant to do so?

Reply:
This still needs to be called (tests fail if it's not called), just with `print_warning=False`